src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/SyntaxTreeExtensions.cs

```diff
@@ -68,7 +68,7 @@ public static bool IsScript(this SyntaxTree syntaxTree)
         {
             Contract.ThrowIfNull(syntaxTree);
 
-            if (position >= syntaxTree.Length)
+            if (position > syntaxTree.Length)
```

Comment:
interesting product bug.  goto-def wouldn't work in c#/vb at the end of a file anyways because of this check.